6-AdvancedScenarios/1-call-api-obo/AppCreationScripts/sample.json

```diff
@@ -5,115 +5,76 @@
         "Client": "React SPA",
         "Service": "Node.js web API",
         "RepositoryUrl": "ms-identity-javascript-react-tutorial",
-        "Endpoint": "AAD v2.0"
+        "Endpoint": "AAD v2.0",
+        "Platform": "javascript",
+        "Languages": [
+            "typescript",
+            "Node"
+        ],
+        "Description": "Handling Conditional Access challenges in an Azure AD protected Node.js web API calling another protected Node.js web API on behalf of a user",
+        "products": ["azure-active-directory", "msal-js", "passport-azure-ad"]
     },
     "AADApps": [
         {
-            "Id": "DownstreamAPI",
-            "Name": "msal-react-downstream",
-            "Kind": "WebApi",
-            "HomePage": "http://localhost:7000/api",
-            "Audience": "AzureADMyOrg",
-            "ManualSteps": [
-                {
-                    "Comment": "Navigate to the Manifest and set 'accessTokenAcceptedVersion' to '2' instead of 'null'"
-                },
-                {
-                    "Comment": "Create a new conditional access policy as described in the sample's README"
-                }
-            ]
-        },
-        {
-            "Id": "MiddletierAPI",
-            "Name": "msal-react-middletier",
+            "Id": "service",
+            "Name": "msal-react-api",
             "Kind": "WebApi",
             "HomePage": "http://localhost:5000/api",
             "Audience": "AzureADMyOrg",
             "PasswordCredentials": "Auto",
+            "SampleSubPath": "6-AdvancedScenarios\\1-call-api-obo\\API",
+            "Scopes": ["access_graph_on_behalf_of_user"],
             "RequiredResourcesAccess": [
                 {
-                    "Resource": "DownstreamAPI",
-                    "DelegatedPermissions": [
-                        "access_downstream_as_user"
-                    ]
+                    "Resource": "Microsoft Graph",
+                    "DelegatedPermissions": ["User.Read", "offline_access"]
                 }
             ],
-            "ManualSteps": [
-                {
-                    "Comment": "Navigate to the Manifest and set 'accessTokenAcceptedVersion' to '2' instead of 'null'"
-                }
-            ]
+            "OptionalClaims": {
+                "AccessTokenClaims": ["idtyp", "xms_cc"]
+            }
         },
         {
-            "Id": "spa",
+            "Id": "client",
             "Name": "msal-react-spa",
             "Kind": "SinglePageApplication",
             "Audience": "AzureADMyOrg",
             "HomePage": "http://localhost:3000/",
-            "ReplyUrls": "http://localhost:3000/",
+            "ReplyUrls": "http://localhost:3000/, http://localhost:3000/redirect.html",
```

Comment:
>http://localhost:3000/redirect.html [](http://example.com/codeflow?start=50&length=35)

I am curious but does one really need to provide path to an entire html as reply Url?

Reply:
This file is located in the[ public directory](https://github.com/Azure-Samples/ms-identity-javascript-react-tutorial/blob/ea768f58578f49053dcff80ed2061c8fe33fa27e/6-AdvancedScenarios/1-call-api-obo/SPA/public/redirect.html) outside the react project routes, so we need a complete file path to redirect to it.